src/wasm-binary.h

```diff
@@ -1195,7 +1195,7 @@ class WasmBinaryBuilder {
   bool debug;
 
   size_t pos = 0;
-  int32_t startIndex = -1;
+  int64_t startIndex = -1;
```

Comment:
Correct me if I'm interpreting the code wrong, but isn't it reading an unsigned 32-bit LEB into a signed 32-bit integer type, which will overflow the upper half of values to negative?